README.md

```diff
@@ -24,6 +24,77 @@ poetry add git+https://github.com/aai-institute/nnbench.git
 
 A PyPI package release is planned for the future.
 
+## Usage Example
+
+Let's say we want to benchmark the following ML model:
+
+```python
+# training.py
+from sklearn.datasets import load_iris
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+
+data = load_iris()
+X, y = data.data, data.target
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3)
+
+model = RandomForestClassifier()
+model.fit(X_train, y_train)
+```
+
+To benchmark your model, you encapsulate the benchmark code into a function and assign the `@benchmark` decorator. This let's our benchmark class collect and later execute the function.
+
+```python
+# benchmarks.py
+import nnbench
+import numpy as np
+from sklearn import base, metrics
+
+
+@nnbench.benchmark(tags=("accuracy",))
+def accuracy(model: base.BaseEstimator, y_test: np.ndarray, y_pred: np.ndarray) -> float:
+    accuracy = metrics.accuracy_score(y_test, y_pred)
+    return accuracy
+```
+
+Now we can instantiate a benchmark runner, to collect, run and report the specified benchmarks. We only execute the benchmarks with the associated tags.
+
+```python
+# run_benchmarks.py
+import pathlib
+from nnbench import runner
+from training import model, X_test, y_test
+
+
+PATH_TO_BENCHMARK_DIR = pathlib.Path('.').resolve()
+
+
+class ExampleRunner(runner.AbstractBenchmarkRunner):
+    # Here you could integrate your custom benchmark handling logic.
+    # We keep it empty in this example
+    pass
```

Comment:
You did the right thing here, but we shouldn't advertise this in the readme. Therefore, we should probably rename the abstract class to just `BenchmarkRunner`.